src/invidious/data_structs/invidious/channel.cr

```diff
@@ -0,0 +1,126 @@
+# Data structs used by Invidious to provide certain features.
+module InvidiousStructs
+  # Struct for representing a cached YouTube channel.
+  #
+  # This is constructed from YouTube's RSS feeds for channels and is
+  # currently only used for storing subscriptions in a user.
+  struct Channel
+    include DB::Serializable
+
+    property id : String
+    property author : String
+    property updated : Time
+    property deleted : Bool
+    # TODO I don't believe the subscripted attribute is actually used.
+    # so this can likely be removed.
+    property subscribed : Time?
```

Comment:
The subscribed property tracks when the channel was subscribed on PubSub, and to resubscribe if it has been more than 4 days.